### PR TITLE
sqlite-vec: init at v0.1.1

### PR DESCRIPTION
**Pull request by anmonteiro**
## Description of changes

https://github.com/asg017/sqlite-vec/releases/tag/v0.1.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sq/sqlite-vec/package.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "vector search SQLite extension that runs anywhere";
+    homepage = "https://github.com/asg017/sqlite-vec";
+    changelog = "https://github.com/asg017/sqlite-vec/releases/tag/v${finalAttrs.version}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ anmonteiro ];
```
Comment (ulrikstrid) [inline pkgs/by-name/sq/sqlite-vec/package.nix:43]: Nested `with` is frowned upon.

**Review by ulrikstrid — CHANGES_REQUESTED** [review]

**Review by ulrikstrid — APPROVED** [review]
With a squash this looks good to go!

File: pkgs/by-name/sq/sqlite-vec/package.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      "dist/libsqlite_vec0${stdenv.hostPlatform.extensions.staticLibrary}" \
+      "dist/vec0${stdenv.hostPlatform.extensions.sharedLibrary}"
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "vector search SQLite extension that runs anywhere";
+    homepage = "https://github.com/asg017/sqlite-vec";
+    changelog = "https://github.com/asg017/sqlite-vec/releases/tag/v${finalAttrs.version}";
```
Comment (afh) [inline pkgs/by-name/sq/sqlite-vec/package.nix:41]: Using `src.rev` here might be more convenient, feel free to leave as is:
```suggestion
    changelog = "https://github.com/asg017/sqlite-vec/releases/tag/${src.rev}";
```

File: pkgs/by-name/sq/sqlite-vec/package.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    install -Dm444 -t "$out/lib" \
+      "dist/libsqlite_vec0${stdenv.hostPlatform.extensions.staticLibrary}" \
+      "dist/vec0${stdenv.hostPlatform.extensions.sharedLibrary}"
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "vector search SQLite extension that runs anywhere";
```
Comment (afh) [inline pkgs/by-name/sq/sqlite-vec/package.nix:39]: According to the [recommendations for meta attributes](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes) the description should start with a capital letter:
```suggestion
    description = "Vector search SQLite extension that runs anywhere";
```

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for this, @anmonteiro.

Please find below minor suggestions for improvement.

**Comment by anmonteiro** [comment]
Thanks @ulrikstrid @afh, addressed review

**Review by ulrikstrid — APPROVED** [review]

**Comment by afh** [comment]
Thanks for addressing the review comments, @anmonteiro, much appreciated. Given that this is a library only package, can you think of a minimal test that could be added as a `checkPhase` to ensure the package works as expected?

**Pull request merged**